cl/recap/utils.py

```diff
@@ -0,0 +1,41 @@
+from typing import Any
+
+from django.forms.models import model_to_dict
+
+from cl.recap.models import PacerFetchQueue
+
+
+def get_court_id_from_fetch_queue(fq: PacerFetchQueue | dict[str, Any]) -> str:
+    """Extracts the court ID from a PacerFetchQueue object or a dictionary.
+
+    This function attempts to retrieve the court ID from the provided
+    PacerFetchQueue object or dictionary. It checks for the court ID in
+    the following order:
+
+    1. From the recap_document's docket_entry's docket.
+    2. From the docket object.
+    3. The top-level `court_id` (for PacerFetchQueue objects) or `court.pk`
+       (for dictionaries).
+
+    :param fq: A PacerFetchQueue object or a dictionary representing a
+    PacerFetchQueue.
+    :return: The court ID as a string.
+    """
+    # Check if the input is a PacerFetchQueue object or a dictionary.
+    is_fetch_queue = isinstance(fq, PacerFetchQueue)
+
+    # Convert PacerFetchQueue to dictionary if necessary.  This allows us to
+    # handle both types consistently.
+    attrs = model_to_dict(fq) if is_fetch_queue else fq
+
+    if attrs.get("recap_document"):
+        rd = fq.recap_document if is_fetch_queue else attrs["recap_document"]
+        court_id = rd.docket_entry.docket.court_id
```

Comment:
I know that you wrote this line in case we ever need to use this method to retrieve the `court_id` from the `recap_document` in the FQ. Currently, this is not required. However, I found that `rd.docket_entry.docket.court_id` will require three additional queries to obtain the `court_id`, and each of these queries will request all the RD, DE, and Docket fields. 
Therefore, it would be better to change the approach for obtaining the `court_id` by using a filtered queryset that retrieves only the required value.

Reply:
Thanks! I'll update the logic